lowranceusr.cc

```diff
@@ -894,18 +1884,75 @@ lowranceusr_route_hdr(const route_head* rte)
   if (global_opts.debug_level >= 1)
     printf(MYNAME " route_hdr: route name \"%s\" num_legs = %d\n",
            qPrintable(rte->rte_name), num_legs);
+}
+
+static void
+lowranceusr4_route_hdr(const route_head* rte)
+{
+  if (global_opts.debug_level >= 1) {
+    printf(MYNAME " writing route #%d (%s) with %d waypts\n",
+           route_uid, qPrintable(rte->rte_name), rte->rte_waypt_ct);
+  }
 
+  /* UID unit number */
+  if (opt_serialnum_i > 0)
+    gbfputint32(opt_serialnum_i, file_out);   // use option serial number if specified
+  else if (rte->fs != (format_specific_data *)NULL)
+    gbfputint32(((lowranceusr4_fsdata *)(rte->fs))->uid_unit, file_out);  // else use serial number from input if valid
+  else
+    gbfputint32(0, file_out);                 // else Write Serial Number = 0
+
+  /* 64-bit UID sequence number */
+  gbfputint32(route_uid++, file_out);
+  gbfputint32(0, file_out);
+
+  /* Route stream version number: seems to be 1 in my data */
+  gbfputint16(1, file_out);
+
+  /* Route name */
+  lowranceusr4_writestr(rte->rte_name, file_out, 2);
+
+  /* Num Legs */
+  gbfputint32(rte->rte_waypt_ct, file_out);
 }
 
 static void
-lowranceusr_track_disp(const Waypoint* wpt)
+lowranceusr4_route_leg_disp(const Waypoint* wpt)
 {
-  if (++trail_point_count <= MAX_TRAIL_POINTS) {
+  for (int i = 0; i < waypt_table_ct; i++) {
+    Waypoint* cmp = waypt_table[i];
+    if (cmp->shortname == wpt->shortname) {
+      lowranceusr4_fsdata *fsdata = (lowranceusr4_fsdata *)cmp->fs;
+      gbfputint32(fsdata->uid_unit, file_out);  // serial number from input if valid
+      gbfputint32(i, file_out); // Sequence Low
+      gbfputint32(0, file_out); // Sequence High
+      if (global_opts.debug_level > 1) {
+        printf(MYNAME " wrote route leg with waypt '%s'\n", qPrintable(wpt->shortname));
+      }
+      break;
+    }
+  }
+}
+
+
+static void
+lowranceusr4_route_trl(const route_head* rte)
```

Comment:
"const route_head* rte" -> "const route_head* /* rte */" or "const route head*" will silence warnings about unused parameter.

Reply:
Thanks for the info.